### PR TITLE
Add regression tests for mrcal

### PR DESCRIPTION
**Pull request by mcm001**

### CODE REVIEW

**Comment by mcm001** [comment]
Not sure why CI just like isnt running at all

**Comment by Gold856** [comment]
Because you've never contributed before

**Comment by mcm001** [comment]
Im attempting to enable the google test plugin, hitting some fun gradle trouble though...

**Comment by mcm001** [comment]
Let me know when this should be rebased again and I'll go ahead and do that, and also add a couple more datasets just for sanity

**Comment by Gold856** [comment]
You'll need this patch to get tests compiling:
```diff
diff --git a/wpical/build.gradle b/wpical/build.gradle
index aa7ef7a18..1c71a66c5 100644
--- a/wpical/build.gradle
+++ b/wpical/build.gradle
@@ -20,8 +20,7 @@ ext {
     nativeName = 'wpical'
     useCpp = true
     sharedCvConfigs = [
-        apriltagDev : [],
-        apriltagTest: []]
+        wpicalTest: []]
     staticCvConfigs = []
 }
 
@@ -33,7 +32,6 @@ def wpilibVersionFileOutput = file("$buildDir/generated/main/cpp/WPILibVersion.c
 
 apply from: "${rootDir}/shared/ceres.gradle"
 apply from: "${rootDir}/shared/opencv.gradle"
-apply from: "${rootDir}/shared/googletest.gradle"
 
 task generateCppVersion() {
     description = 'Generates the wpilib version class'
@@ -189,8 +187,6 @@ model {
                         it.linker.args << '-lGL'
                     }
                 }
-
-                nativeUtils.useRequiredLibrary(it, 'opencv_shared');
             }
         }
     }
@@ -213,6 +209,25 @@ model {
                     }
                 }
             }
+            binaries.all {
+                if (it.targetPlatform.name == nativeUtils.wpi.platforms.roborio || it.targetPlatform.name.startsWith("linuxarm")) {
+                    it.buildable = false
+                    return
+                }
+                lib project: ':wpiutil', library: 'wpiutil', linkage: 'static'
+                lib project: ':wpigui', library: 'wpigui', linkage: 'static'
+                lib project: ':thirdparty:imgui_suite', library: 'imgui', linkage: 'static'
+                lib project: ':apriltag', library: 'apriltag', linkage: 'static'
+                nativeUtils.useRequiredLibrary(it, 'ceres')
+                nativeUtils.useRequiredLibrary(it, 'opencv_static')
+                it.cppCompiler.define 'GLOG_USE_GLOG_EXPORT'
+                if (it.targetPlatform.operatingSystem.isWindows()) {
+                    it.cppCompiler.define('GLOG_DEPRECATED', '__declspec(deprecated)')
+                } else {
+                    it.cppCompiler.define('GLOG_DEPRECATED', '[[deprecated]]')
+                }
+                lib project: ':thirdparty:googletest', library: 'googletest', linkage: 'static'
+            }
         }
     }
 }
```
There are a few warnings and build errors still when the tests are being built. Since the icon is broken and breaks the build, you'll have to replace the wpical icon with something like the SysId icon (you can commit it if you want and I can revert it on my own.)

File: wpical/build.gradle [hunk 1]
```diff
@@ -17,6 +18,11 @@ if (OperatingSystem.current().isWindows()) {
 
 ext {
     nativeName = 'wpical'
+    useCpp = true
+    sharedCvConfigs = [
+        apriltagDev : [],
+        apriltagTest: []]
```
Comment (Gold856) [inline wpical/build.gradle]:
```suggestion
        wpicalTest: []]
```

File: wpical/build.gradle [hunk 2]
```diff
@@ -27,6 +33,7 @@ def wpilibVersionFileOutput = file("$buildDir/generated/main/cpp/WPILibVersion.c
 
 apply from: "${rootDir}/shared/ceres.gradle"
 apply from: "${rootDir}/shared/opencv.gradle"
+apply from: "${rootDir}/shared/googletest.gradle"
```
Comment (Gold856) [inline wpical/build.gradle]: Remove
```suggestion
```

File: wpical/build.gradle [hunk 3]
```diff
@@ -187,6 +192,27 @@ model {
...
+                cpp {
+                    source {
+                        srcDirs 'src/test/native/cpp'
+                        include '**/*.cpp'
+                    }
+                    exportedHeaders {
+                        srcDirs 'src/test/native/include', 'src/main/native/cpp'
+                    }
+                }
+            }
```
Comment (Gold856) [inline wpical/build.gradle:211]:
```suggestion
            }
            binaries.all {
                if (it.targetPlatform.name == nativeUtils.wpi.platforms.roborio || it.targetPlatform.name.startsWith("linuxarm")) {
                    it.buildable = false
                    return
                }
                lib project: ':wpiutil', library: 'wpiutil', linkage: 'static'
                lib project: ':wpigui', library: 'wpigui', linkage: 'static'
                lib project: ':thirdparty:imgui_suite', library: 'imgui', linkage: 'static'
                lib project: ':apriltag', library: 'apriltag', linkage: 'static'
                nativeUtils.useRequiredLibrary(it, 'ceres')
                nativeUtils.useRequiredLibrary(it, 'opencv_static')
                it.cppCompiler.define 'GLOG_USE_GLOG_EXPORT'
                if (it.targetPlatform.operatingSystem.isWindows()) {
                    it.cppCompiler.define('GLOG_DEPRECATED', '__declspec(deprecated)')
                } else {
                    it.cppCompiler.define('GLOG_DEPRECATED', '[[deprecated]]')
                }
                lib project: ':thirdparty:googletest', library: 'googletest', linkage: 'static'
            }
```

**Review by Gold856 — CHANGES_REQUESTED** [review]

**Pull request merged**